libnavigation-core/src/main/java/com/mapbox/navigation/core/reroute/RerouteController.kt

```diff
@@ -115,6 +116,8 @@ sealed class RerouteState {
 
     /**
      * Route has been fetched.
+     *
+     * @param routerOrigin which router was used to fetch the route
      */
-    object RouteFetched : RerouteState()
+    data class RouteFetched(val routerOrigin: RouterOrigin) : RerouteState()
```

Comment:
`RouterOrigin` is added to `RouteFetched` state. It looks reasonable, as the same way the `RouteOrigin` is added to `List<RouteFailure>` of `Failed` state